mypyc/genops.py

```diff
@@ -1164,25 +1215,28 @@ def handle_ext_method(self, cdef: ClassDef, fdef: FuncDef) -> None:
 
         # If this overrides a parent class method with a different type, we need
         # to generate a glue method to mediate between them.
-        for cls in class_ir.mro[1:]:
-            if (name in cls.method_decls and name != '__init__'
+        for base in class_ir.mro[1:]:
+            if (name in base.method_decls and name != '__init__'
                     and not is_same_method_signature(class_ir.method_decls[name].sig,
-                                                     cls.method_decls[name].sig)):
+                                                     base.method_decls[name].sig)):
 
                 # TODO: Support contravariant subtyping in the input argument for
                 # property setters. Need to make a special glue method for handling this,
                 # similar to gen_glue_property.
 
-                if fdef.is_property:
-                    f = self.gen_glue_property(cls.method_decls[name].sig, func_ir, class_ir,
-                                               cls, fdef.line)
-                else:
-                    f = self.gen_glue_method(cls.method_decls[name].sig, func_ir, class_ir,
-                                             cls, fdef.line)
-
-                class_ir.glue_methods[(cls, name)] = f
+                f = self.gen_glue(base.method_decls[name].sig, func_ir, class_ir, base, fdef)
+                class_ir.glue_methods[(base, name)] = f
                 self.functions.append(f)
 
+        # If the class allows interpreted children, create glue
+        # methods that dispatch via the Python API. These will go in a
+        # "shadow vtable" that will be assigned to interpreted
+        # children.
+        if class_ir.allow_interpreted_children:
+            f = self.gen_glue(func_ir.sig, func_ir, class_ir, class_ir, fdef, do_py_ops=True)
+            class_ir.glue_methods[(class_ir, name)] = f
```

Comment:
What if a class that allow interpreted children has a native subclass? Will these glue methods cause trouble?

I wonder if putting the python glue methods in a separate data structure would be clearer.

Reply:
Shouldn't cause trouble